src/modifiers.rs

```diff
@@ -29,3 +33,26 @@ impl StatusModifier for StateFromPeopleNowPresent {
     }
 }
 
+/// This modifier adds internal version information to the output.
+pub struct LibraryVersions;
+
+impl StatusModifier for LibraryVersions {
+    fn modify(&self, status: &mut api::Status) {
+        // Add library version information
+        let api_version = api::get_version().to_string();
+        let server_version = ::get_version().to_string();
+
+        // Create version map if it doesn't exist yet
+        if status.ext_versions.is_absent() {
```

Comment:
And besides all the technicalities, `if status.ext_versions.is_absent()` is *way* easier to read than `if let Optional::Absent = status.ext_versions` :)